src/Workspaces/Core/Portable/FindSymbols/Shared/AbstractSyntaxIndex_Persistence.cs

```diff
@@ -0,0 +1,168 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.Host;
+using Microsoft.CodeAnalysis.Internal.Log;
+using Microsoft.CodeAnalysis.Shared.Extensions;
+using Microsoft.CodeAnalysis.Shared.Utilities;
+using Microsoft.CodeAnalysis.Storage;
+using Roslyn.Utilities;
+
+namespace Microsoft.CodeAnalysis.FindSymbols
+{
+    internal partial class AbstractSyntaxIndex<TIndex> : IObjectWritable
+    {
+        private static readonly string s_persistenceName = typeof(TIndex).Name;
+        private static readonly Checksum s_serializationFormatChecksum = Checksum.Create("27");
```

Comment:
this value incremented as well to ensure that the new indices don't collide with the old ones.

Reply:
Is it possible in the future that the serialization of the top level and other index will change separately?

Reply:
yes.  this si a good point.  with the current approach, we'll invalidate both indices and regenerate them.  but it's not the end of the world to do that :)